packages/integration-sdk-cli/src/commands/options.ts

```diff
@@ -1,51 +1,93 @@
+    '-p, --project-path <directory>',
+    'path to integration project directory',
+    process.cwd(),
+  );
+}
 
 /**
- * Determines the JupiterOne `apiBaseUrl` based on these mutually-exclusive `options`:
- * - --api-base-url - a specific URL to use
- * - --development - use the development API
- *
- * @throws Error if both --api-base-url and --development are specified
+ * Configure the filesystem interaction code to function against `${options.projectPath}/.j1-integration by setting the
+ * global `process.env.JUPITERONE_INTEGRATION_STORAGE_DIRECTORY` variable.
  */
-export function getApiBaseUrlOption(options: any): string {
-  let apiBaseUrl: string;
-  if (options.apiBaseUrl) {
-    if (options.development) {
-      throw new Error(
-        'Invalid configuration supplied.  Cannot specify both --api-base-url and --development(-d) flags.',
-      );
-    }
-    apiBaseUrl = options.apiBaseUrl;
-  } else {
-    apiBaseUrl = getApiBaseUrl({
-      dev: options.development,
-    });
-  }
-  return apiBaseUrl;
+export function configureRuntimeFilesystem(options: PathOptions): void {
+  process.env.JUPITERONE_INTEGRATION_STORAGE_DIRECTORY = path.resolve(
+    options.projectPath,
+    '.j1-integration',
+  );
 }
 
-interface SynchronizationJobSourceOptions {
+export interface SyncOptions {
   source: 'integration-managed' | 'integration-external' | 'api';
-  scope?: string;
-  integrationInstanceId?: string;
+  scope?: string | undefined;
+  integrationInstanceId?: string | undefined;
+  uploadBatchSize: number;
+  uploadRelationshipBatchSize: number;
+  skipFinalize: boolean;
+}
+
+export function addSyncOptionsToCommand(command: Command): Command {
+  return command
+    .option(
+      '-i, --integrationInstanceId <id>',
+      '_integrationInstanceId assigned to uploaded entities and relationships',
+    )
+    .option(
+      '--source <integration-managed|integration-external|api>',
+      'specify synchronization job source value',
+      (value: string) => {
+        if (
+          value !== 'integration-managed' &&
+          value !== 'integration-external' &&
+          value !== 'api'
+        ) {
+          throw new Error(
+            `--source must be one of "integration-managed", "integration-external", or "api"`,
+          );
+        } else {
+          return value;
+        }
+      },
+      'integration-managed',
+    )
+    .option('--scope <anystring>', 'specify synchronization job scope value')
+    .option(
+      '-u, --upload-batch-size <number>',
+      'specify number of items per batch for upload',
+      (value, _previous: Number) => Number(value),
+      DEFAULT_UPLOAD_BATCH_SIZE,
+    )
+    .option(
+      '-ur, --upload-relationship-batch-size <number>',
+      'specify number of relationships per batch for upload',
+      (value, _previous: Number) => Number(value),
+      DEFAULT_UPLOAD_BATCH_SIZE,
+    )
+    .option('--skip-finalize', 'skip synchronization finalization', false);
```

Comment:
consider adding what the end result of this is. Something like: The Sync job will remain open to receive additional data.